modules/platform/nuxeo-platform-document-routing/nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/io/TaskWriter.java

```diff
@@ -108,7 +113,12 @@ public void writeEntityBody(Task item, JsonGenerator jg) throws IOException {
             if (StringUtils.isNotBlank(workflowInstanceId)) {
                 NodeAccessRunner nodeAccessRunner = new NodeAccessRunner(wrapper.getSession(), workflowInstanceId,
                         nodeId);
-                nodeAccessRunner.runUnrestricted();
+                try {
+                    nodeAccessRunner.runUnrestricted();
```

Comment:
Why not fixing the root cause in `NodeAccessRunner` instead of catching?

Reply:
As we discussed on slack the root cause is a DocumentNotFound exception we can only catch where we need to.

We have identified other issues as well: the NodeAccessRunner is also called by `org.nuxeo.ecm.platform.routing.core.io.TaskCompletionRequestJsonReader.readEntity(JsonNode)`

When we list the tasks, corrupted tasks show, but obviously miss information, which we may want to avoid in the UI.

Here is a sample response with a first corrupted task and a well formed task:
https://jira.nuxeo.com/browse/NXP-28798?focusedCommentId=453928&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-453928